Framework/Utils/test/test_RootTreeWriterWorkflow.cxx

```diff
@@ -0,0 +1,118 @@
+#include <vector>
+
+using namespace o2::framework;
+
+#define ASSERT_ERROR(condition)                                   \
+  if ((condition) == false) {                                     \
+    LOG(ERROR) << R"(Test condition ")" #condition R"(" failed)"; \
+  }
+
+constexpr int kTreeSize = 10; // elemants to send and write
+DataProcessorSpec getSourceSpec()
+{
+  auto initFct = [](InitContext& ic) {
+    auto counter = std::make_shared<int>();
+    *counter = 0;
+
+    auto processingFct = [counter](ProcessingContext& pc) {
+      if (*counter >= kTreeSize) {
+        // don't publish more
+        return;
+      }
+      o2::test::Polymorphic a(*counter);
+      pc.outputs().snapshot(OutputRef{ "output" }, a);
+      int& metadata = pc.outputs().make<int>(Output{ "TST", "METADATA", 0, Lifetime::Timeframe });
+      metadata = *counter;
+      *counter = *counter + 1;
+    };
+
+    return processingFct;
+  };
+  return DataProcessorSpec{ "source", // name of the processor
+                            {},
+                            { OutputSpec{ { "output" }, "TST", "SOMEOBJECT", 0, Lifetime::Timeframe },
+                              OutputSpec{ { "meta" }, "TST", "METADATA", 0, Lifetime::Timeframe } },
+                            AlgorithmSpec(initFct) };
+}
+
+// this is the direct definition of the processor spec, can be generated from
+// MakeRootTreeWriterSpec as shown below
+DataProcessorSpec getSinkSpec()
+{
+  auto initFct = [](InitContext& ic) {
+    std::string fileName = gSystem->TempDirectory();
+    fileName += "/test_RootTreeWriter.root";
+    using WriterType = RootTreeWriter<o2::test::Polymorphic, int>;
+    auto writer = std::make_shared<WriterType>(fileName.c_str(),      // output file name
+                                               "testtree",            // tree name
+                                               "input", "polyobject", // input key and branch name
+                                               "meta", "counter"      // input key and branch name
+                                               );
+    auto counter = std::make_shared<int>();
+    *counter = 0;
+
+    // the callback to be set as hook at stop of processing for the framework
+    auto finishWriting = [writer]() { writer->close(); };
+    ic.services().get<CallbackService>().set(CallbackService::Id::Stop, finishWriting);
+
+    auto processingFct = [writer, counter](ProcessingContext& pc) {
+      (*writer)(pc);
+      *counter = *counter + 1;
+      if (*counter >= kTreeSize) {
+        pc.services().get<ControlService>().readyToQuit(true);
+      }
+    };
+
+    return processingFct;
+  };
+
+  return DataProcessorSpec{ "sink",
+                            { InputSpec{ "input", "TST", "SOMEOBJECT" }, //
+                              InputSpec{ "meta", "TST", "METADATA" } },  //
+                            Outputs{},
+                            AlgorithmSpec(initFct) };
+}
+
+void defineDataProcessing(WorkflowSpec& specs)
```

Comment:
@ktf, just noticed that this will interfere with #1086, both in the newly added and in the moved file. I suggest to merge this PR only after #1086